.github/workflows/macos_tests.yml

```diff
@@ -38,7 +44,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          brew install fftw libomp python3
+          brew install fftw llvm python3
```

Comment:
Is installing llvm now necessary? What's the problem with new apple clang?

Reply:
Yeah. So my current understanding: 
* `llvm` was not (and still is not) necessary for Xcode 13. Because it's included.
* `llvm` is not included in Xcode 14 (‽) <sup>[Citation very much needed]</sup>. I *have* confirmed that it doesn't work without.
* `libomp` was never needed on it's own because it's included with `llvm` libraries.
* No need to point CMake to LIBCXX because it can find it by itself. And that path was wrong for Xcode 14.

Regarding the last point, it might  *never* have been needed to be pointed to LIBCXX on these runners. Just necessary on M1/2. And I from which I copied the build procedure from my laptop.

Reply:
I would be happy if I'm wrong and this can be simplified even further, but unless I'm missing a trick...

Reply:
> llvm is not included in Xcode 14 (‽) [Citation very much needed]. I have confirmed that it doesn't work without.

Yeah, I'm not convinced that's true, the images definitely come with apple clang available out-of-the-box: https://github.com/actions/runner-images/blob/2aec524970584138c5986109ca7be7ae89b0899d/images/macos/macos-12-Readme.md#language-and-runtime. Note also that vanilla clang from homebrew should already be available (but under a different path).

I'd just like to understand what's going on exactly if possible, because if there is something wrong with apple clang we should either fix it or note it in the documentation.

Reply:
On my dev branch:
* 4f8d351 [fails, can't find OpenMP](https://github.com/UCL/TDMS/actions/runs/3190080975)
* c52b4b8 [also fails, can't find OpenMP](https://github.com/UCL/TDMS/actions/runs/3190153934)

Reply:
Hmm. OK so it seems I can use the already-installed llvm's libomp via:
```
$(brew --prefix llvm@14)/lib
```

But this is hard-coding a "14". I think I'd rather run
```
brew install llvm
```
and use either same line without specifying `@14`.

Reply:
It's also consistent with the Mac-specific instructions.